src/ntapfuse_ops.c

```diff
@@ -0,0 +1,389 @@
+
+  //log_file_op("Mkdir",fpath,0,BlockSize);
+
+  return mkdir (fpath, mode | S_IFDIR) ? -errno : 0;
+}
+
+int
+ntapfuse_unlink (const char *path)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  return unlink (fpath) ? -errno : 0;
+}
+
+int
+ntapfuse_rmdir (const char *path)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  size_t size = getDirSize(fpath);
+
+  //log_file_op("Rmdir",fpath,size,size);
+
+  return rmdir (fpath) ? -errno : 0;
+}
+
+int
+ntapfuse_symlink (const char *path, const char *link)
+{
+  char flink[PATH_MAX];
+  fullpath (link, flink);
+
+  return symlink (path, flink) ? -errno : 0;
+}
+
+int
+ntapfuse_rename (const char *src, const char *dst)
+{
+  char fsrc[PATH_MAX];
+  fullpath (src, fsrc);
+
+  char fdst[PATH_MAX];
+  fullpath (dst, fdst);
+
+  return rename (fsrc, fdst) ? -errno : 0;
+}
+
+int
+ntapfuse_link (const char *src, const char *dst)
+{
+  char fsrc[PATH_MAX];
+  fullpath (src, fsrc);
+
+  char fdst[PATH_MAX];
+  fullpath (dst, fdst);
+
+  return link (fsrc, fdst) ? -errno : 0;
+}
+
+int
+ntapfuse_chmod (const char *path, mode_t mode)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  return chmod (fpath, mode) ? -errno : 0;
+}
+
+int
+ntapfuse_chown (const char *path, uid_t uid, gid_t gid)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  return chown (fpath, uid, gid) ? -errno : 0;
+}
+
+int
+ntapfuse_truncate (const char *path, off_t off)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  return truncate (fpath, off) ? -errno : 0;
+}
+
+int
+ntapfuse_utime (const char *path, struct utimbuf *buf)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  return utime (fpath, buf) ? -errno : 0;
+}
+
+int
+ntapfuse_open (const char *path, struct fuse_file_info *fi)
+{
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  int fh = open (fpath, fi->flags);
+  if (fh < 0)
+    return -errno;
+
+  fi->fh = fh;
+
+  return 0;
+}
+
+char* addquote(char* str){
+  char* newstr = calloc(1,strlen(str)+2);
+  *newstr='\'';
+  strcat(newstr,str);
+  char* t = "\'";
+  strcat(newstr,t);
+  return newstr;
+}
+
+
+
+int
+ntapfuse_read (const char *path, char *buf, size_t size, off_t off,
+	   struct fuse_file_info *fi)
+{
+
+  char fpath[PATH_MAX];
+  fullpath (path, fpath);
+
+  
+
+  return pread (fi->fh, buf, size, off) < 0 ? -errno : size;
+}
+
+
+
+int
+ntapfuse_write (const char *path, const char *buf, size_t size, off_t off,
+	    struct fuse_file_info *fi)
+{
+  char fpath[PATH_MAX];
+  int res;
+  int initFileSize=0;
+  int usage=0;
+  FILE *f;
+
+  fullpath (path, fpath);
+
+  if (newfile==1) { /* if file is new */
+      usage=size<BLOCK_SIZE?BLOCK_SIZE:size;
+      newfile=0;
+  } else {  /* if the file already exist */
+      f = fopen(fpath, "r");
+      fseek(f, 0, SEEK_END); 
+      initFileSize = ftell(f);
+      fclose(f);
+      
+      if(initFileSize<BLOCK_SIZE) 
+              usage = initFileSize+size>BLOCK_SIZE?initFileSize+size-BLOCK_SIZE:0;
```

Comment:
Enclose in {}
Maintain consistency with indentation.  It might be the editor thing; set tab spaces to 4 or other values to be consistent within the file.

Reply:
fixed, thanks